src/SFML/Window/Unix/WindowImplX11.cpp

```diff
@@ -91,7 +91,8 @@ constexpr unsigned long eventMask = FocusChangeMask | ButtonPressMask | ButtonRe
 constexpr unsigned int maxTrialsCount = 5;
 
 // Filter the events received by windows (only allow those matching a specific window)
-Bool checkEvent(::Display*, XEvent* event, XPointer userData)
+// NOLINTNEXTLINE(readability-non-const-parameter)
+Bool checkEvent(::Display*, XEvent* event, const XPointer userData)
```

Comment:
Why do we add an exception here? I don't see how this is different from any other parameter.

Same above

Reply:
`clang-tidy` changed `userData` to `const` in the first pass, then kept coming back and complaining that `userData` could be `const`.

So I kept it `const` and added the exception to shut it up.

Reply:
This also confused me so I played with it locally and found no way to satisfy clang-tidy without replacing `XPointer` with something else but then we lose some expressiveness so I didn't want to do that. Turning it off here seemed reasonable.

Reply:
> `clang-tidy` changed `userData` to `const` in the first pass

Would it then not make more sense to disable the first lint, and not annotate the parameter `const`?

Reply:
> > `clang-tidy` changed `userData` to `const` in the first pass
> 
> Would it then not make more sense to disable the first lint, and not annotate the parameter `const`?

No, I think it's useful to declare the function's intent to not modify `userData`

Reply:
> No, I think it's useful to declare the function's intent to not modify `userData`. Achieving some measure of [const-correctness](https://isocpp.org/wiki/faq/const-correctness) is always worthwhile.

Following this argumentation, the function should look like this:
```cpp
Bool checkEvent(::Display* const, XEvent* const event, const XPointer userData);
```

To clarify, we're talking about _top-level_ `const` here, it doesn't express if you have read-only access through the pointer. All it does is protect you from doing `userData = somethingElse` in the function, i.e. reassigning the parameter.

My point is that we don't use this convention in SFML, so now having a single parameter out of 3 marked `const` raises more questions than it answers.

Reply:
I'm fine with both options so I'll defer to what Bromeon wants.

Reply:
> To clarify, we're talking about _top-level_ `const` here

Point taken; the rest of the `const` in this PR were applied to functions, not function arguments